core/src/main/scala/io/chrisdavenport/github/internals/RequestConstructor.scala

```diff
@@ -46,44 +47,49 @@ object RequestConstructor {
     c.expect[B](req)
   }
 
+  private val RE_LINK: Regex = "[\\s]*<(.*)>; rel=\"(.*)\"".r
+
   private def getNextUri[F[_]: MonadError[*[_], Throwable]](r: Response[F]): Option[Uri] = {
-    for {
-      next <- r.headers.toList.map(Link.matchHeader).collect{
-        case Some(Link(uri, Some("next"), _, _, _)) => uri
-      }.headOption
-    } yield next
+    r.headers.toList.flatMap {
+      case Header(name, value) if name == "Link".ci => value.split(",").toList
+      case _ => Nil
+    }.collectFirstSome {
+      case RE_LINK(uri, "next") => Uri.fromString(uri).toOption
+      case _ => None
+    }
   } 
 
   def runPaginatedRequest[F[_]: Sync, B: EntityDecoder[F, ?]](
     auth: Option[Auth],
     extendedUri: Uri,
   ):  Kleisli[Stream[F, ?], Client[F], B] = Kleisli{ c =>
     val uri = Uri.resolve(baseUrl(auth), extendedUri)
-
-    unfoldLoopEval(uri){uri => 
+    unfoldLoopEval(uri){ uri =>
       val baseReq = Request[F](method = Method.GET, uri = uri)
         .withHeaders(extraHeaders)
       val req = auth.fold(baseReq)(setAuth(_)(baseReq))
-      c.run(req).use{resp => 
-          resp.as[B].map{
+      c.fetch(req) { resp =>
```

Comment:
Why did we switch from `run` to `fetch`. `run` is clearer use semantics in my opinion.

Reply:
No clear reason - I just didn't know how to do it using f.
But looking at it now, I think it should also work using `use` as well.
I guess this part can be changed back - I can do it if you want.